sagemathinc/cocalc
# Track anonymous account conversions

# Description
- [x] Expose `central_log` as *write only*
- [x] All queries are automatically prepended with "webapp-"
- [x] Add 2 tracking events 
    - [x] Sign up with email
    - [x] Sign up with passport
- [x] Add query to determine the number of de-anonymized accounts

```sql
SELECT COUNT(id) 
FROM central_log 
WHERE time >= NOW() - interval '24 hours' 
AND (event = 'webapp-email_sign_up' OR event = 'webapp-add_passport') 
AND value@> '{"source": "anonymous_account"}';
```

# Testing Steps
1. Create an anonymous account
1. Sign up with email
1. Run the sql query above. The count should be 1
1. Create another anonymous account.
1. Sign up with a passport
1. The query count should be 2.
1. Create an account using the old method (just sign out and navigate to `/app`)
1. The query count should still be 2.

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
